polaris-react/src/components/IndexTable/IndexTable.scss

```diff
@@ -85,6 +85,12 @@ $loading-panel-height: 53px;
   width: 100%;
   min-width: 100%;
   border-collapse: collapse;
+
+  &:not(.Table-select-mode) {
```

Comment:
Kind of weird adding a class for negative styles but I don't dislike this pattern. The alternative would be to add the radii and overflow to .Table and cancel them out with .Table-select-mode